### ISSUE TITLE
[data grid] Column header labels still abbreviated after Autosize Columns

### ISSUE THREAD

**Issue by AndreasDBS** [mui/mui-x#15388]
### The problem in depth

We are building an enterprise web app that has lots of tables, and a table may have 20 or more columns. We want columns to be as wide as required, but not wider than that. Most importantly: column headers must always be readable, no matter if the sort order has changed and the sort icon is displayed.

Here comes the problem: take your excellent documentation as an example:
[docs](https://mui.com/x/react-data-grid/column-dimensions/#autosizing)
First, you click on Autosize Columns in that example (the Include Headers option is ticked). You can see that as a result the Brand ID column is given a width which nicely fits the Brand ID label into the header, it's not too wide and not too narrow.

Now, choose to sort for the Brand ID column. The column header now displays the Sort-Arrow icon to indicate the ordering. However, that Icon overlaps the column header label, which is then abbreviated with "...". A subsequent use of the Autosize Columns button does not fix that.

We never want column header labels to be abbreviated. So, here is the question:

1. Is it possible to disable the abbreviation of column header labels?
2. Is it possible to somehow change the algorithm that calculates the width style attribute for column headers? We know which columns are used for ordering and could adjust the width with that information. There just need to be a way to hook into that algorithm. Is there a way for this?

Latest version tested: Mui Data Grid v7.22.2.

### Your environment

<details>
  <summary>`npx @mui/envinfo`</summary>

  System:
    OS: Windows 10 10.0.19045
  Binaries:
    Node: 20.17.0
    npm: 10.8.2
  npmPackages:
    @emotion/react: ^11.13.3 => 11.13.3
    @emotion/styled: ^11.13.0 => 11.13.0
    @mui/base:  5.0.0-beta.58
    @mui/core-downloads-tracker:  6.1.3
    @mui/icons-material: ^6.1.3 => 6.1.3
    @mui/lab: ^6.0.0-beta.11 => 6.0.0-beta.11
    @mui/material: ^6.1.3 => 6.1.3
    @mui/private-theming:  6.1.3
    @mui/styled-engine:  6.1.3
    @mui/system:  6.1.3
    @mui/types:  7.2.18
    @mui/utils:  6.1.3
    @mui/x-charts: ^7.20.0 => 7.20.0
    @mui/x-charts-vendor:  7.20.0
    @mui/x-data-grid:  7.20.0
    @mui/x-data-grid-pro: ^7.20.0 => 7.20.0
    @mui/x-date-pickers:  7.20.0
    @mui/x-date-pickers-pro: ^7.20.0 => 7.20.0
    @mui/x-internals:  7.20.0
    @mui/x-license: ^7.20.0 => 7.20.0
    @types/react: ^18.2.14 => 18.2.15
    react: ^18.2.0 => 18.2.0
    react-dom: ^18.2.0 => 18.2.0
    typescript: ^5.0.2 => 5.1.6
</details>

**Search keywords**: autoresize column width
**Order ID**: 79546

**Comment by michelengelen** [comment]
cc @romgrk since it might be related to virtualization ☝🏼

**Comment by AndreasDBS** [comment]
@michelengelen Thanks a lot to confirm that and for your quick response. I was already trying to look into that useGridColumnResize.js but am more than happy to hand this over to you. Cheers!

**Comment by AndreasDBS** [comment]
Sorry, I just wanted to comment but pressed "Close with comment" by mistake.

**Comment by michelengelen** [comment]
> So, **github-project-automation** changed this issue from **Needs refinement** to **Backlog** - what's going on here, is this issue still in the pipeline to get fixed, or is this issue lost due to my closing it by mistake? Should I enter a new one?

Moved it back to the 'needs refinement' lane. Thanks for the heads up! 👍🏼

**Comment by AndreasDBS** [comment]
Hi guys, just wanted to say, I know you have a lot to do, and even though we have licensed your lib we can't demand anything.
I am really of fan of your work, but I think it is just fair to tell you that I can't further persuade my boss to ship our product with your datagrid with this simple bug.

As simple as this bug might look - we just cannot ship a product which hides what information is shown in columns.

**Comment by michelengelen** [comment]
> Hi guys, just wanted to say, I know you have a lot to do, and even though we have licensed your lib we can't demand anything. I am really of fan of your work, but I think it is just fair to tell you that I can't further persuade my boss to ship our product with your datagrid with this simple bug.
> 
> As simple as this bug might look - we just cannot ship a product which hides what information is shown in columns.

Fair enough ... If it helps I can have a look today and try to fix before the release today.

**Comment by AndreasDBS** [comment]
That would be absolutely amazing, thank you!

**Comment by michelengelen** [comment]
Hey @AndreasDBS ... I found a solution, but it is not super ideal. Opened a PR anyways with backporting to v7. Lets wait for the reviews and go from there. Is that OK for you?
